components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -962,19 +962,33 @@ func (p *pg) CreateRule(ctx context.Context, rule *v2.Rule) (*v2.Rule, error) {
 	if err != nil {
 		return nil, p.processError(err)
 	}
+	
+	row:= tx.QueryRowContext(ctx,
+		`SELECT query_rule_table_associations($1);`, rule.ID)
+	if err != nil {
+		return nil, p.processError(err)
+	}
+	var associations []string
+	if err := row.Scan(pq.Array(&associations)); err != nil {
+		return nil, p.processError(err)
+	}
+	// If any associations return, then the rule already exists in current, staged, or both tables
+	if len(associations) > 0 {
```

Comment:
We could maybe try to do this check in the database with a trigger? Thoughts?

Reply:
:+1: I think a trigger is a good idea here. Forbid adding to `iam_staged_rule_conditions` if a rule with that ID exists in either one of the two tables 🤔